compact_str/src/repr/mod.rs

```diff
@@ -452,6 +439,36 @@ impl Repr {
         len == 0
     }
 
+    /// Return a reference to the underlying buffer
+    #[inline]
+    pub fn as_ptr(&self) -> *const u8 {
+        // initially has the value of the stack pointer, conditionally becomes the heap pointer
+        let mut pointer = ensure_read_ptr(self as *const Self as *const u8);
+        let heap_pointer = ensure_read_ptr(self.0 as *const u8);
+        if self.last_byte() >= HEAP_MASK {
+            pointer = heap_pointer;
+        }
+        pointer
+    }
+
+    /// Return a mutable reference to the underlying buffer
+    ///
+    /// # Safety
+    /// Must only be called on a dynamically allocated [`Repr`].
```

Comment:
This should be "not static repr", since it only asserts on non-static repr

Reply:
My reasoning was flawed anyway. I changed `as_mut_ptr()`'s implementation, so it will automatically allocated writable data if it previously contained immutable data.